lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -364,6 +337,125 @@ void FirRegLower::initialise(OpBuilder &regBuilder, OpBuilder &initBuilder,
   recurse(type, "");
 }
 
+bool FirRegLower::isInitialisePreset(sv::RegOp reg) {
+  return module->hasAttr("firrtl.random_init_width") &&
+         reg->hasAttr("firrtl.random_init_start") &&
+         reg->hasAttr("firrtl.random_init_end");
+}
+
+void FirRegLower::initialisePreset(OpBuilder &regBuilder,
+                                   OpBuilder &initBuilder, sv::RegOp reg) {
+  // Extract the required random initialisation width and compute the number of
+  // `RANDOM calls required to fill it up, as well as the final register width.
+  auto randomInitWidthAttr =
+      module->getAttrOfType<IntegerAttr>("firrtl.random_init_width");
+  assert(randomInitWidthAttr &&
+         "firrtl.random_init_width required for preset initialisation");
+
+  uint64_t randomInitWidth = randomInitWidthAttr.getUInt();
+  uint64_t numRandomSources = llvm::divideCeil(randomInitWidth, randomWidth);
+  uint64_t randomRegWidth = randomWidth * numRandomSources;
```

Comment:
Can `randomInitWidth` ever be zero? I'm not sure. Doing an early exit if so should work.

Reply:
The pass that creates the attribute in the first place won't if it's zero. I'll add an assert that it's non-zero.